pandas/core/arrays/categorical.py

```diff
@@ -53,6 +53,11 @@
 
 
 def _cat_compare_op(op):
+
+    # Note: using unpack_and_defer here doesn't break any tests, but the
+    #  behavior here is idiosyncratic enough that I'm not confident enough
+    #  to change it.
+    # @ops.unpack_and_defer
```

Comment:
@TomAugspurger are you the right person to ask about Categorical methods?  In particular, why does this only return NotImplemented for ABCSeries and not for ABCDataFrame and ABCIndexClass?

Reply:
No idea. Returning NotImplemented seems reasonable, as long as it ends up back here with the unboxed values.

Reply:
Categorical == DataFrame is weird both before and after

```
data = ["a", "b", 2, "a"]
cat = pd.Categorical(data)
idx = pd.CategoricalIndex(cat)
ser = pd.Series(cat)
df = pd.DataFrame(cat)
```

master
```
>>> cat == df
array([[ True, False, False,  True],
       [False,  True, False, False],
       [False, False,  True, False],
       [ True, False, False,  True]])
```

PR
```
>>> cat == df
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/ops.py", line 2127, in f
    other = _align_method_FRAME(self, other, axis=None)
  File "pandas/core/ops.py", line 2021, in _align_method_FRAME
    right = to_series(right)
  File "pandas/core/ops.py", line 1983, in to_series
    given_len=len(right)))
ValueError: Unable to coerce to Series, length must be 1: given 4
```

If we transpose then we get all-True, the only difference being that in the PR the result is a DataFrame instead of an ndarray.